File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int levels(TreeNode* root){
        if(root==NULL) return 0;
        return 1 + max(levels(root->left), levels(root->right));
    }
    int bfs(TreeNode* root, int n , int &sum){
        if(root==NULL) return 0;
        if(n==1){
            sum+=(root->val);
        }
        bfs(root->left, n-1, sum);
        bfs(root->right, n-1, sum);
        return sum;
    }
    int maxLevelSum(TreeNode* root) {
        int level = levels(root);
        int maxSum = INT_MIN;
        int ans =0;
        for(int i = 1;i<=level;i++){
            int sum = 0;
            bfs(root, i, sum); //root, level, sum
            if(sum>maxSum){
                maxSum = sum;
                ans = i;
            }
        }
        return ans;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int sum = 0;
    public int levels(TreeNode root){
        if(root==null) return 0;
        return 1+ Math.max( levels(root.left), levels(root.right));
    }
    public void bfs(TreeNode root, int n){
        if(root== null) return;
        if(n==1){
            this.sum+=(root.val);
        }
        bfs(root.left, n-1);
        bfs(root.right, n-1);
    }
    public int maxLevelSum(TreeNode root) {
        int level = levels(root);

        int maxSum = Integer.MIN_VALUE;
        int ans = 1; //level of max sum

        for(int i = 1; i<=level; i++){
            sum = 0;
            bfs(root, i);
            if(sum>maxSum){
                maxSum = sum;
                ans = i;
            }
        }
        return ans;
    }
}